slack_bolt/listener_matcher/builtins.py

```diff
@@ -188,18 +205,58 @@ def func(body: Dict[str, Any]) -> bool:
     return build_listener_matcher(func, asyncio, base_logger)
 
 
-def workflow_step_execute(
+# -------------
+# functions
+
+
+def function_action(
     callback_id: Union[str, Pattern],
+    constraints: Union[str, Pattern, Dict[str, Union[str, Pattern]]],
+    asyncio: bool = False,
+    base_logger: Optional[Logger] = None,
+) -> Union[ListenerMatcher, "AsyncListenerMatcher"]:
+    if isinstance(constraints, (str, Pattern)):
+
+        def func(body: Dict[str, Any]) -> bool:
+            return _function_block_action(callback_id, constraints, body)
+
+        return build_listener_matcher(func, asyncio, base_logger)
+
+    elif "type" in constraints:
+        action_type = constraints["type"]
+        if action_type == "block_actions":
+            return function_block_action(callback_id, constraints, asyncio)
```

Comment:
should asyncio be passed in here? same for the next conditional down

Reply:
Good catch 💯  It is not used in this PR but it should be used in a PR coming soon, if it is not it will be removed